glean-core/src/lib.rs

```diff
@@ -51,6 +49,21 @@ lazy_static! {
         Uuid::parse_str("c0ffeec0-ffee-c0ff-eec0-ffeec0ffeec0").unwrap();
 }
 
+/// The Glean configuration.
+///
+/// Optional values will be filled in with default values.
+#[derive(Debug, Clone)]
+pub struct Configuration {
+    /// Whether upload should be enabled.
+    pub upload_enabled: bool,
+    /// Path to a directory to store all data in.
+    pub data_path: String,
+    /// The application ID (will be sanitized during initialization).
+    pub application_id: String,
+    /// The maximum number of events to store before sending a ping.
```

Comment:
nit: is that specific to the 'events' ping or to all pings having events? I'd call that out, here.